framework/helpers/BaseHtml.php

```diff
@@ -1719,7 +1719,8 @@ public static function renderSelectOptions($selection, $items, &$tagOptions = []
         );
         $options['encode'] = ArrayHelper::getValue($options, 'encode', $encode);
 
-        if (null !== ($prompt = ArrayHelper::remove($tagOptions, 'prompt'))) {
+        $prompt = ArrayHelper::remove($tagOptions, 'prompt');
+        if (null !== $prompt) {
```

Comment:
Sure it is. Else there would be no such style. The thing is in Yii we're primarily using normal order and want consistency.